src/Collections-Strings-Tests/StringTest.class.st

```diff
@@ -447,11 +447,11 @@ StringTest >> testAsHTMLString [
 { #category : 'tests - converting' }
 StringTest >> testAsInteger [
 
-	self assert: '1796exportFixes-tkMX' asInteger equals: 1796.
-	self assert: 'donald' asInteger isNil.
-	self assert: 'abc234def567' asInteger equals: 234.
+	self should: [ '1796exportFixes-tkMX' asInteger ] raise: Error.
+	self should: [ 'donald' asInteger ] raise: Error.
+	self should: [ 'abc234def567' asInteger ] raise: Error.
```

Comment:
Please add a case 

```Smalltalk
'123 4' asInteger 
```
as this shows clearly what can happens if for instance in an edit field someone uses accidentially a blank. 
In existing Pharo 12 this is (unfortunately) accepted and returns 123 without an exception.